Sources/TestingMacros/CMakeLists.txt

```diff
@@ -0,0 +1,64 @@
+# This source file is part of the Swift.org open source project
+#
+# Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+# Licensed under Apache License v2.0 with Runtime Library Exception
+#
+# See http://swift.org/LICENSE.txt for license information
+# See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+
+cmake_minimum_required(VERSION 3.28)
+
+if(POLICY CMP0157)
+  cmake_policy(SET CMP0157 NEW)
+endif()
+
+project(TestingMacros
+  LANGUAGES Swift)
+
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake)
+
+add_compile_definitions(_CRT_SECURE_NO_WARNINGS)
+include(FetchContent)
+set(FETCHCONTENT_BASE_DIR ${CMAKE_BINARY_DIR}/_d)
+FetchContent_Declare(SwiftSyntax
+  GIT_REPOSITORY https://github.com/apple/swift-syntax
+  GIT_TAG main)
+FetchContent_MakeAvailable(SwiftSyntax)
+
+include(Settings)
+add_executable(TestingMacros
+  ConditionMacro.swift
+  SuiteDeclarationMacro.swift
+  Support/Additions/DeclGroupSyntaxAdditions.swift
+  Support/Additions/EditorPlaceholderExprSyntaxAdditions.swift
+  Support/Additions/FunctionDeclSyntaxAdditions.swift
+  Support/Additions/MacroExpansionContextAdditions.swift
+  Support/Additions/TokenSyntaxAdditions.swift
+  Support/Additions/TriviaPieceAdditions.swift
+  Support/Additions/TypeSyntaxProtocolAdditions.swift
+  Support/Additions/VersionTupleSyntaxAdditions.swift
+  Support/Additions/WithAttributesSyntaxAdditions.swift
+  Support/Argument.swift
+  Support/AttributeDiscovery.swift
+  Support/AvailabilityGuards.swift
+  Support/CommentParsing.swift
+  Support/ConditionArgumentParsing.swift
+  Support/CRC32.swift
+  Support/DiagnosticMessage.swift
+  Support/DiagnosticMessage+Diagnosing.swift
+  Support/SourceCodeCapturing.swift
+  Support/SourceLocationGeneration.swift
+  TagMacro.swift
+  TestDeclarationMacro.swift
+  TestingMacrosMain.swift)
+set_target_properties(TestingMacros PROPERTIES
+  ENABLE_EXPORTS TRUE)
+target_compile_options(TestingMacros PRIVATE -parse-as-library)
```

Comment:
Curious why this is necessary also

Reply:
Ah, actually `-parse-as-library` may indeed be appropriate because we use `@main`. If so, this makes sense and is fine

Reply:
Your analysis is correct.  Without it I get:
```
LINK : fatal error LNK1561: entry point must be defined
```